test/integration/framework/controlplane_utils.go

```diff
@@ -233,7 +233,10 @@ func startAPIServerOrDie(controlPlaneConfig *controlplane.Config, incomingServer
 	}
 	var lastHealthContent []byte
 	err = wait.PollImmediate(100*time.Millisecond, 30*time.Second, func() (bool, error) {
-		result := privilegedClient.Get().AbsPath("/healthz").Do(context.TODO())
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
```

Comment:
100ms seems short... I think dial timeout defaults to 10 seconds or so... anything on the order of seconds would be an improvement over the current non-terminating request

Reply:
I changed the timeout to `5s` (i was thinking along the line that `/health` probe should be really fast returning in the order of milliseconds, so the test can be more strict in asserting it, but maybe this is not the right place to enforce or report on `/health` probe latency)